jsonschema/_validators.py

```diff
@@ -190,9 +190,14 @@ def enum(validator, enums, instance, schema):
 
 
 def ref(validator, ref, instance, schema):
-    with validator.resolver.resolving(ref) as resolved:
+    scope, resolved = validator.resolver.resolve(ref)
+    validator.resolver.push_scope(scope)
```

Comment:
The performance improvement here was to remove the contextmanager.  I'm not sure how to compose over that change. The interfaces seem incompatible to me (but I could be missing something).

Falling back to the old interface would be easy enough:
```python

if hasattr(validator.resolver, 'resolve'):
    ... # do what this does now
    return

... # Use the old contextmanager way

```